commit 5e6b4d1f9540bb7b75705faf80e412fc0c66fe84
Author: GeorgeClark <GeorgeClark@0b4bb1d4-4e5a-0410-9cc4-b2b747904278>
Date:   Mon Nov 5 05:07:25 2012 +0000

    Item11267: Don't use "HEAD" to detect pseudo install.
    Note:  This patch fixes bug reported in Item12225
    
    A real, non-pseudo-installed extension will crash configure if a perl
    version object is compared to an alpha string.
    
    9999.99_999 will be used to indicate a pseudo-installed release.
    
    git-svn-id: http://svn.foswiki.org/trunk@15909 0b4bb1d4-4e5a-0410-9cc4-b2b747904278

~~~PATCH 76e28354522a6d6cccc76c66f99d2424:24ce3ef562c0021782b103c81774b980 lib/Foswiki/Configure/UIs/EXTENSIONS.pm (Foswiki-1.1.0,Foswiki-1.1.1,Foswiki-1.1.2,Foswiki-1.1.3,Foswiki-1.1.4,Foswiki-1.1.5)
--- lib/Foswiki/Configure/UIs/EXTENSIONS.pm	2012-11-07 23:45:21.269274066 -0500
+++ lib/Foswiki/Configure/UIs/EXTENSIONS.pm.new	2012-11-07 23:46:39.844344573 -0500
@@ -280,7 +280,7 @@
         if ( $ext->{installedRelease} ) {
 
             # The module is installed; check the version
-            if ( $ext->{installedVersion} eq 'HEAD' ) {
+            if ( $ext->{installedVersion} eq '9999.99_999' ) {
 
                 # pseudo-installed
                 $install = 'pseudo-installed';

~~~PATCH 491da6d7e2ec5695532924a322d00775:62615300a753905267c1620321a1c51e lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.0)
--- lib/Foswiki/Configure/Dependency.pm 2010-10-04 11:26:34.000000000 -0400
+++ lib/Foswiki/Configure/Dependency.pm.new     2012-11-08 18:06:36.866582897 -0500
@@ -218,7 +218,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }

~~~PATCH 761e2ab62309739b740cb506bd6f2cc3:0276c444cd7d5068434f7be5c88a7e6d lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.1)
--- lib/Foswiki/Configure/Dependency.pm 2010-10-25 17:58:13.000000000 -0400
+++ lib/Foswiki/Configure/Dependency.pm.new     2012-11-08 18:09:42.117464357 -0500
@@ -218,7 +218,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }

~~~PATCH b865e17f72f28a16cf39a7f0af0f8ddc:74546e2a4f1d06582d53b780a641c509 lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.2)
--- lib/Foswiki/Configure/Dependency.pm 2010-11-09 19:29:12.000000000 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new     2012-11-08 18:12:03.809843937 -0500
@@ -218,7 +218,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }

~~~PATCH 6f2c6eb77ffbc95a4068e584bd5d1aa5:f446c7779d79cb9b9a59fa4cf62cd657 lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.3)
--- lib/Foswiki/Configure/Dependency.pm 2011-04-16 16:29:33.000000000 -0400
+++ lib/Foswiki/Configure/Dependency.pm.new     2012-11-08 18:15:42.067347880 -0500
@@ -220,7 +220,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }

~~~PATCH 4e525d29a245785ae810316835c18e62:221ccfdb31d41b8d376a5e095003e88d lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.4)
--- lib/Foswiki/Configure/Dependency.pm 2011-12-20 10:33:22.000000000 -0500
+++ lib/Foswiki/Configure/Dependency.pm 2012-11-08 17:05:55.982213647 -0500
@@ -220,7 +220,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }

~~~PATCH fdeeb7f236608b7792ad0845bf2279f9:8da49fcc5b7babb0661a1ec2979a03c7 lib/Foswiki/Configure/Dependency.pm (Foswiki-1.1.5)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-07 23:46:56.378148979 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-07 23:46:14.091649218 -0500
@@ -220,7 +220,7 @@
             if ( -l "$dir/$path" ) {
 
                 # Assume pseudo-installed
-                $this->{installedVersion} = 'HEAD';
+                $this->{installedVersion} = '9999.99_999';
             }
             last;
         }


